daft/io/common.py

```diff
@@ -44,6 +45,48 @@ def _get_tabular_files_scan(
     else:
         raise NotImplementedError(f"Tabular scan with config not implemented: {storage_config.config}")
 
+    ### FEATURE_FLAG: $DAFT_V2_SCANS
+    #
+    # This environment variable will make Daft use the new "v2 scans" and MicroPartitions when building Daft logical plans
+    if os.getenv("DAFT_V2_SCANS", "0") == "1":
+        assert (
+            os.getenv("DAFT_MICROPARTITIONS", "0") == "1"
+        ), "DAFT_V2_SCANS=1 requires DAFT_MICROPARTITIONS=1 to be set as well"
+
+        scan_op: ScanOperatorHandle
+        if isinstance(path, list):
+            # Eagerly globs each path and fallback to AnonymousScanOperator.
+            # NOTE: We could instead have GlobScanOperator take a list of paths and mux the glob output streams
+            runner_io = get_context().runner().runner_io()
+            file_infos = runner_io.glob_paths_details(path, file_format_config=file_format_config, io_config=io_config)
+
+            # Infer schema if no hints provided
+            inferred_or_provided_schema = (
+                schema_hint
+                if schema_hint is not None
+                else runner_io.get_schema_from_first_filepath(file_infos, file_format_config, storage_config)
+            )
+
+            scan_op = ScanOperatorHandle.anonymous_scan(
+                file_infos.file_paths, inferred_or_provided_schema._schema, file_format_config, storage_config
+            )
+        elif isinstance(path, str):
+            scan_op = ScanOperatorHandle.glob_scan(
```

Comment:
We wouldn't need both branches if `glob_scan` took in a list of paths?

Reply:
Yes, I just didn't want to implement the new logic/API just yet. I can create a ticket for cleanup

Reply:
See: https://github.com/Eventual-Inc/Daft/pull/1577